Source/Particles/ParticleBoundaries.H

```diff
@@ -28,13 +30,37 @@ struct ParticleBoundaries
 
     bool CheckAll (ParticleBoundaryType bc);
 
+    void BuildReflectionModelParsers ();
+
     ParticleBoundaryType xmin_bc;
     ParticleBoundaryType xmax_bc;
     ParticleBoundaryType ymin_bc;
     ParticleBoundaryType ymax_bc;
     ParticleBoundaryType zmin_bc;
     ParticleBoundaryType zmax_bc;
 
+    // reflection models for absorbing boundaries
+    std::string reflection_model_xlo_str = "0.0";
+    std::string reflection_model_xhi_str = "0.0";
+    std::string reflection_model_ylo_str = "0.0";
+    std::string reflection_model_yhi_str = "0.0";
+    std::string reflection_model_zlo_str = "0.0";
+    std::string reflection_model_zhi_str = "0.0";
+
+    std::unique_ptr<amrex::Parser> reflection_model_xlo_parser;
+    std::unique_ptr<amrex::Parser> reflection_model_xhi_parser;
+    std::unique_ptr<amrex::Parser> reflection_model_ylo_parser;
+    std::unique_ptr<amrex::Parser> reflection_model_yhi_parser;
+    std::unique_ptr<amrex::Parser> reflection_model_zlo_parser;
+    std::unique_ptr<amrex::Parser> reflection_model_zhi_parser;
```

Comment:
`std::unique_ptr` doesn't have a copy constructor, so that's why the default copy ctor for `ParticleBoundaries` is now `deleted`. We can work out how to handle these